src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -5867,6 +5875,30 @@ private bool HasNoUserControlProperties(XamlObjectDefinition objectDefinition)
 			return null;
 		}
 
+		/// <summary>
+		/// Builds the x:Name cache for faster lookups in <see cref="FindSubElementByName(XamlObjectDefinition, string)"/>.
+		/// </summary>
+		/// <remarks>
+		/// The lookup is performed by searching for the x:Name, then determine if one of the ancestors
+		/// is known. This avoids doing linear lookups at many levels recusively for the same nodes.
+		/// </remarks>
+		private void BuildNameCache(XamlObjectDefinition topLevelControl)
+		{
+			foreach (var element in EnumerateSubElements(topLevelControl))
+			{
+				var nameMember = FindMember(element, "Name");
+
+				if (nameMember?.Value is string name)
+				{
+					if (!_nameCache.TryGetValue(name, out var list))
```

Comment:
`TryAdd` should be faster
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.tryadd?view=net-6.0

Reply:
I wish we could use it... :) It's not in netstandard2.0...

Reply:
Just curious, won't `TryAdd(name, new())` always allocate, and the allocation might be unnecessary?

With `TryGetValue`, we are going to allocate only if `name` doesn't exist in the cache.

Reply:
good catch, what I had in mind was `CollectionsMarshal.GetValueRefOrAddDefault` but that's (likely) not in `netstandard2.0` either